packages/gatsby/src/utils/babel-loader-helpers.ts

```diff
@@ -98,17 +121,34 @@ const prepareOptions = (babel, options = {}, resolve = require.resolve) => {
   ]
 }
 
-const mergeConfigItemOptions = ({ items, itemToMerge, type, babel }) => {
+type mergeConfigItemOptionsType = ({
+  items,
+  itemToMerge,
+  type,
+  babel,
+}: {
+  items: ConfigItem[]
+  itemToMerge: ConfigItem
+  type: CreateConfigItemOptions["type"]
+  babel: typeof Babel
+}) => ConfigItem[]
+
+export const mergeConfigItemOptions: mergeConfigItemOptionsType = ({
+  items,
+  itemToMerge,
+  type,
+  babel,
+}) => {
   const index = _.findIndex(
     items,
-    i => i.file.resolved === itemToMerge.file.resolved
+    i => i.file!.resolved === itemToMerge.file!.resolved
```

Comment:
Can you use optional chaining rather than not-null assertion here and elsewhere, so avoid potential runtime errors